### PR TITLE
Support customization wether use short or long names

### LINKED ISSUE #34: Add option to use bool instead of boolean
As bool would be the actual type hint in PHP7 we should allow this to be configured to be the default when inferring from things other than type hints

### PR DESCRIPTION
**Pull request by jens1o**
Does it look good? If yes, I make the tests for TypeUtil.ts.

Fixes #34

### CODE REVIEW

**Comment by jens1o** [comment]
@neild3r Should I fix the tests, or change the default way?

**Comment by neild3r** [comment]
Can we invert the config value to useShortNames and set the default to true, then just ensure the output of the original tests match up with the default. Then you can add the tests for the type util.

**Comment by jens1o** [comment]
Okay, I'll do that.

File: src/block/function.ts [hunk 1]
```diff
@@ -65,9 +66,13 @@ export default class FunctionBlock extends Block
     public getReturnFromName(name:string):string
     {
         if (/^(is|has)/.test(name)) {
-            return 'boolean';
+            return TypeUtil.getFormattedTypeByName('bool');
         }
 
+        // if (/^get/.test(name)) {
+        //     return 'mixed';
```
Comment (neild3r) [inline src/block/function.ts]: Can we just take this out of this PR completely, I was debating adding this previously, regardless really it belongs in it's own PR

Comment (jens1o) [inline src/block/function.ts]: Yeah.

**Comment by jens1o** [comment]
The 2 failing tests are looking to be false, or what did I overlooked? @neild3r

**Comment by neild3r** [comment]
You missed some param data types in the fixtures

**Comment by jens1o** [comment]
I'm sorry for the question: But how can I add my test to the testing suite?

**Comment by jens1o** [comment]
May I ask again why the tests I added are failing? I had cleared the cache, so it should work... I'm somewhat blind. Sorry.

**Comment by neild3r** [comment]
Not sure but I would probably set the value on rather than trying to manipulate the configuration object from vscode. 

Really this is why static's suck I always try to avoid them as they make testing crappy. Really if we wanted to start doing things like this we should use a container. However as I'm not that much of a JS/TS dev I've not used may container/DI patterns in TS. However this is part of a broader development that we should probably save until we look at doing multiple languages or something.

You could switch to singleton pattern that allows you to construct instances or set configs separately just for testing.

**Comment by jens1o** [comment]
Okay, will do that asap. And yeah, I'm also not a pure js/ts dev, I like php ❤️

**Comment by jens1o** [comment]
Fine with this @neild3r ?

File: src/util/TypeUtil.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    private static _instance: TypeUtil;
+
+    /**
+     * Holds wether we use short names or not
+     *
+     * @type {bool|null}
+     */
+    private _useShortNames: any;
+
+    private constructor() {
```
Comment (neild3r) [inline src/util/TypeUtil.ts]: If you make this public then you can just create an instance for each test instead of relying on the singleton, does enforce the singleton less but I think it's worth it for nicer testing.

I would also consider getting rid of the constructor altogether and create a getter for the useShortNames that when it's null will access the config and set it. This way it only accesses the config if it needs to.

Comment (jens1o) [inline src/util/TypeUtil.ts]: Okay, I'll do that.

**Comment by jens1o** [comment]
like this? 👍

**Comment by neild3r** [comment]
Yes like that, looks like it isn't calling the method properly when retrieving the config value though normally.

screen shot 2017-05-25 at 11 41 15

**Comment by jens1o** [comment]
Can Typescript handle `===`? That was something php-ish.

**Comment by jens1o** [comment]
Let's try that.

**Comment by jens1o** [comment]
Seems to work. Finally ❤️

**Comment by neild3r** [comment]
Hurray, I'll merge in later.

**Comment by jens1o** [comment]
Do you schedule an update soon? I need this feature to improve my workflow 😆

**Comment by neild3r** [comment]
Yes, I'll try to do one tonight I'm just messing around with #47 at the moment
